test/rules/prefer-conditional/test.ts.lint

```diff
@@ -0,0 +1,37 @@
+let x;
+if (true) {
+~~ [X]
+    x = 1;
+} else {
+    x = 2;
+}
+
+if (true)
+~~ [X]
+    x = "TRUE";
```

Comment:
The rule needs `x = <<code>>` *exactly*, it doesn't just search to see if it appears somewhere. So normally that would filter out complex logic.
Maybe you were thinking of something like this?

```ts
let x: number[];
if (so) {
	x = [1,2,3].map(n => {
		switch (n) {
			... lots of code ...
		}
	});
} else {
	x = [];
}
```

Maybe the rule could only activate if the expression takes up a single line.